src/Api/Controllers/OrganizationSponsorshipsController.cs

```diff
@@ -0,0 +1,207 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Bit.Core.Context;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.Models.Api;
+using Bit.Core.Models.Api.Request;
+using Bit.Core.Models.Table;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Utilities;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Bit.Api.Controllers
+{
+    [Route("organization/sponsorship")]
+    [Authorize("Application")]
+    public class OrganizationSponsorshipsController : Controller
+    {
+        private readonly IOrganizationSponsorshipService _organizationsSponsorshipService;
+        private readonly IOrganizationSponsorshipRepository _organizationSponsorshipRepository;
+        private readonly IOrganizationRepository _organizationRepository;
+        private readonly IOrganizationUserRepository _organizationUserRepository;
+        private readonly ICurrentContext _currentContext;
+        private readonly IUserService _userService;
+
+        public OrganizationSponsorshipsController(IOrganizationSponsorshipService organizationSponsorshipService,
+            IOrganizationSponsorshipRepository organizationSponsorshipRepository,
+            IOrganizationRepository organizationRepository,
+            IOrganizationUserRepository organizationUserRepository,
+            IUserService userService,
+            ICurrentContext currentContext)
+        {
+            _organizationsSponsorshipService = organizationSponsorshipService;
+            _organizationSponsorshipRepository = organizationSponsorshipRepository;
+            _organizationRepository = organizationRepository;
+            _organizationUserRepository = organizationUserRepository;
+            _userService = userService;
+            _currentContext = currentContext;
+        }
+
+        [HttpPost("{sponsoringOrgId}/families-for-enterprise")]
+        [SelfHosted(NotSelfHostedOnly = true)]
+        public async Task CreateSponsorship(Guid sponsoringOrgId, [FromBody] OrganizationSponsorshipRequestModel model)
+        {
+            var requiredSponsoringProductType = StaticStore.GetSponsoredPlan(model.PlanSponsorshipType)?.SponsoringProductType;
+            var sponsoringOrg = await _organizationRepository.GetByIdAsync(sponsoringOrgId);
+            if (requiredSponsoringProductType == null ||
+                sponsoringOrg == null ||
+                StaticStore.GetPlan(sponsoringOrg.PlanType).Product != requiredSponsoringProductType.Value)
+            {
+                throw new BadRequestException("Specified Organization cannot sponsor other organizations.");
```

Comment:
This is domain-specific business logic relating to things like business rules and authorization. Adding those things to the underlying "service" will allow this API to run in parallel with other technologies (eg. adding GRPC, GraphQL) without having to duplicate these validations. It will also make sure that if someone would make these calls from another controller, the data would still be kept safe because the rules would still be enforced.

Reply:
@coroiu We were just having an internal discussion about where it should live to. I'm going to move it to the services as well for exactly the reasons you brought up.  Thanks for chiming in!

Reply:
Oh cool! Very new to the code base so I might as well have been waaaay off! Thanks for the reponse! :D